compiler/passes/normalize.cpp

```diff
@@ -1751,14 +1751,15 @@ static void change_method_into_constructor(FnSymbol* fn) {
 
   // Now check that the function name matches the name of the type
   // attached to 'this'.
```

Comment:
I am not proposing defining a “string type”.  I am proposing defining
an Identifier type.




On Jun 16, 2016, at 7:32 AM, Michael Ferguson <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/passes/normalize.cpp<https://github.com/chapel-lang/chapel/pull/4032#discussion_r67355327>:

> @@ -1751,14 +1751,15 @@ static void change_method_into_constructor(FnSymbol* fn) {
>
>    // Now check that the function name matches the name of the type
>    // attached to 'this'.


Yes, I think that if we declared our own string type (that amounts to const char* for now, but ensures that the value has been unique'd with astr) we could easily support == and even support it in the future if we changed our mind about the giant string hashtable uniquing all strings.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/chapel-lang/chapel/pull/4032/files/7eec1be31971474efd4fc442f9090d3f38669cc0#r67355327>, or mute the thread<https://github.com/notifications/unsubscribe/ACERpqQuLioBJ0HHPhLTxXwWlcZ3qf5rks5qMV50gaJpZM4I1y5G>.